tla-io/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala

```diff
@@ -119,19 +119,12 @@ class SanyParserPassImpl @Inject() (
           }
         }
 
+        rootModule.map { m => nextPass.updateModule(this, None, m) }
         true
     }
   }
 
-  /**
-   * Get the next pass in the chain. What is the next pass is up
-   * to the module configuration and the pass outcome.
-   *
-   * @return the next pass, if exists, or None otherwise
-   */
-  override def next(): Option[Pass] =
-    rootModule map { m =>
-      nextPass.setModule(m)
-      nextPass
-    }
+  override def dependencies = Set()
+
+  override def transformations = Set(ModuleProperty.Parsed)
```

Comment:
You can probably just remove ModuleProperty.Parsed from the enum. Parsing is a special pass, it has to be first, and obviously the whole tool won't work if the module doesn't parse. And there's no way to un-parse a module.